desc/integrals/singularities.py

```diff
@@ -546,24 +543,13 @@ def eval_pt(eval_data_i):
                 st,
                 sz,
             )
-            return jnp.sum(k * (w * (1 - eta))[..., jnp.newaxis], axis=1)
+            return jnp.sum(k * (w * e)[..., jnp.newaxis], axis=1)
 
-        f += batch_map(eval_pt, eval_data, chunk_size).reshape(
+        return batch_map(eval_pt, eval_data, chunk_size).reshape(
             eval_grid.num_nodes, kernel.ndim
         )
-        return f, source_data
-
-    # This error should be raised earlier since this is not the only place
-    # we need the higher dummy NFP value, but the error message is more
-    # helpful with the nfp loop docstring.
-    errorif(
-        source_grid.num_zeta == 1 and source_grid.NFP == 1,
-        msg="Source grid cannot compute toroidal effects.\n"
-        "Increase NFP of source grid to e.g. 64.\n"
-        "This is required to " + nfp_loop.__doc__,
-    )
-    f = jnp.zeros((eval_grid.num_nodes, kernel.ndim))
-    f, _ = fori_loop(0, source_grid.NFP, nfp_loop, (f, source_data))
```

Comment:
this seems to work without passing source_data in the f_data tuple. pointing out in case I'm missing something about pure functions